db/migrate/db_migrate_20200513201952_rename_source_type_ddf_strings_v2_rb.rb

```diff
@@ -0,0 +1,60 @@
+class RenameSourceTypeDdfStringsV2 < ActiveRecord::Migration[5.2]
+  DDF_V2_MAPPINGS = [
+    ["required-validator", "required"],
+    ["pattern-validator", "pattern"],
+    ["min-length-validator", "min-length"],
+    ["url-validator", "url"],
+    ["switch-field", "switch"]
+  ]
+
+  def up
+    rename(DDF_V2_MAPPINGS)
+  end
+
+  def down
+    rename(DDF_V2_MAPPINGS.map(&:reverse))
+  end
+
+  private
+
+  def rename(mappings)
+    SourceType.all.each do |st|
+      schema = st.schema.try(:with_indifferent_access)
+      next unless schema
```

Comment:
Is there any way that if `schema` isn't created and we skip it, that we skip a record we needed? 

Should we bail completely if `schema.nil?`

Reply:
I think it's just normal - there are some SourceTypes that don't have schemas, this is on CI:

```
irb(main):001:0> SourceType.where(:schema => nil).map(&:product_name)
  SourceType Load (0.5ms)  SELECT "source_types".* FROM "source_types" WHERE "source_types"."schema" IS NULL
{"@timestamp":"2020-05-14T20:33:40.148967 ","hostname":"sources-api-946-njsm6","pid":66580,"tid":"2afa9f5545b4","level":"debug","message":"  SourceType Load (0.5ms)  SELECT \"source_types\".* FROM \"source_types\" WHERE \"source_types\".\"schema\" IS NULL"}
=> ["Red Hat OpenStack", "Red Hat Virtualization", "Red Hat CloudForms", "VMware vSphere"]
```

Lines up with the `seeds.rb` file:
OpenStack:
https://github.com/RedHatInsights/sources-api/blob/a4c2e9368e91e3b7458dd002fc7c419193a9ecac/db/seeds/source_types.yml#L220-L223
RHEV:
https://github.com/RedHatInsights/sources-api/blob/a4c2e9368e91e3b7458dd002fc7c419193a9ecac/db/seeds/source_types.yml#L224-L227
CF:
https://github.com/RedHatInsights/sources-api/blob/a4c2e9368e91e3b7458dd002fc7c419193a9ecac/db/seeds/source_types.yml#L170-L172
vSphere:
https://github.com/RedHatInsights/sources-api/blob/a4c2e9368e91e3b7458dd002fc7c419193a9ecac/db/seeds/source_types.yml#L259-L262

Reply:
Cool. Thanks.